### PR TITLE
fix Product Commerce component not displaying issue

### PR DESCRIPTION
**Pull request by cyrusxzw**
Ticket:
https://aesoponline.atlassian.net/browse/CON-622

It is a quick fix which will has minimum impact on the components using 'useOnScreen' hook. Ideal solution is to rebuild  'useOnScreen' hook by using callback ref. But if so, will need to pass function rather than ref that will be a huge work. 

ps: QA will test all the related components to make sure no breaks.

### CODE REVIEW

File: src/customHooks/useOnScreen/useOnScreen.ts [hunk 1]
```diff
@@ -20,39 +20,40 @@ const useOnScreen: UseOnScreen = (
...
-        },
-      );
-
-      observer.observe(currentRef);
-    }
-
-    return function cleanup() {
-      if (hasIntersectionObserver) {
-        observer.unobserve(currentRef);
+    setTimeout(() => {
```
Comment (MkMan) [inline src/customHooks/useOnScreen/useOnScreen.ts:23]: What does wrapping it in `setTimeout` do?

File: src/customHooks/useOnScreen/useOnScreen.ts [hunk 2]
```diff
@@ -20,39 +20,40 @@ const useOnScreen: UseOnScreen = (
...
+          {
+            rootMargin,
+            threshold,
+          },
+        );
+
+        observer.observe(currentRef);
       }
-    };
+      return function cleanup() {
```
Comment (MkMan) [inline src/customHooks/useOnScreen/useOnScreen.ts:51]: Can you please keep the line break above the cleanup function.

Comment (cyrusxzw) [reply src/customHooks/useOnScreen/useOnScreen.ts:51]: yep

File: src/customHooks/useOnScreen/useOnScreen.ts (hunk 1 above)
Comment (cyrusxzw) [reply src/customHooks/useOnScreen/useOnScreen.ts:23]: To get ref after page content load.

**Review by MkMan — APPROVED** [review]

**Pull request merged**